--- origsrc/harfbuzz-0.9.9/configure.ac	2012-12-05 16:36:04.000000000 -0600
+++ src/harfbuzz-0.9.9/configure.ac	2012-12-16 18:53:38.202588400 -0600
@@ -88,14 +88,21 @@ if test "x$GCC" = "xyes"; then
 	esac
 fi
 
+hb_plat_win32=no
 hb_os_win32=no
 AC_MSG_CHECKING([for native Win32])
 case "$host" in
+  *-*-cygwin*)
+    hb_plat_win32=yes
+    hb_os_win32=no
+    ;;
   *-*-mingw*)
+    hb_plat_win32=yes
     hb_os_win32=yes
     ;;
 esac
 AC_MSG_RESULT([$hb_os_win32])
+AM_CONDITIONAL(PLATFORM_WIN32, test "$hb_plat_win32" = "yes")
 AM_CONDITIONAL(OS_WIN32, test "$hb_os_win32" = "yes")
 
 have_pthread=false
--- origsrc/harfbuzz-0.9.9/src/Makefile.am	2012-11-30 01:05:32.000000000 -0600
+++ src/harfbuzz-0.9.9/src/Makefile.am	2012-12-16 18:54:27.783424300 -0600
@@ -193,6 +193,8 @@ DIST_SUBDIRS += hb-ucdn
 if OS_WIN32
 export_symbols = -export-symbols harfbuzz.def
 harfbuzz_def_dependency = harfbuzz.def
+endif
+if PLATFORM_WIN32
 libharfbuzz_la_LINK = $(CXXLINK) $(libharfbuzz_la_LDFLAGS)
 else
 # Use a C linker, not C++; Don't link to libstdc++
